packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,584 @@
+import { Transaction } from "bitcoinjs-lib-v6";
+import { Network } from "@caravan/bitcoin";
+import { UTXO, AnalyzerOptions, FeeBumpStrategy, Satoshis } from "./types";
+import {
+  BtcTxComponent,
+  BtcTxInputTemplate,
+  BtcTxOutputTemplate,
+} from "./btcTransactionComponents";
+import { getOutputAddress, estimateTransactionVsize } from "./utils";
+import BigNumber from "bignumber.js";
+
+/**
+ * TransactionAnalyzer Class
+ *
+ * This class provides comprehensive analysis of Bitcoin transactions, including
+ * fee estimation, RBF (Replace-By-Fee) and CPFP (Child-Pays-For-Parent) capabilities.
+ * It's designed to help wallet developers make informed decisions about fee bumping
+ * strategies for unconfirmed transactions.
+ *
+ * Key Features:
+ * - Analyzes transaction inputs, outputs, fees, and size
+ * - Determines RBF and CPFP eligibility
+ * - Recommends optimal fee bumping strategy
+ * - Estimates fees for RBF and CPFP operations
+ * - Provides detailed transaction information for wallet integration
+ *
+ * Usage:
+ * const analyzer = new TransactionAnalyzer(txHex, options);
+ * const analysis = analyzer.analyze();
+ *
+ * @class
+ */
+export class TransactionAnalyzer {
+  private readonly _originalTx: Transaction;
+  private readonly _network: Network;
+  private readonly _targetFeeRate: number;
+  private readonly _absoluteFee: BigNumber;
+  private readonly _availableUtxos: UTXO[];
+  private readonly _changeOutputIndex: number | undefined;
+  private readonly _incrementalRelayFee: BigNumber;
+  private readonly _requiredSigners: number;
+  private readonly _totalSigners: number;
+  private readonly _addressType: string;
```

Comment:
I'd expect this to match the actual address/script types we have that can be defined using an export from caravan/bitcoin I think. basically, we wouldn't accept just ANY string as valid here.

Reply:
added new [type](https://github.com/caravan-bitcoin/caravan/pull/114/commits/8da49623b7f7ede0c72ea4fd31a4847d0811293e#diff-b0c4ef170382d75023cd41fe14a8a588936bcc592b31cb86b8abc3178065254fR64)